src/js/components/Search.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import sourceStore from '../stores/SourceStore';
+
+export default class Search extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      sources: sourceStore.getSources(),
+      searchInput: ''
+    };
+    this.handleSearch = this.handleSearch.bind(this);
```

Comment:
don't bind here. refer to comment about bind in `src/js/components/Header.jsx` file above.